src/Network/Freddy.hs

```diff
@@ -0,0 +1,48 @@
+{-# OPTIONS -XOverloadedStrings #-}
+module Network.Freddy (connect, respondTo) where
+
+import Network.AMQP
+import Data.Text (Text, pack)
+import Data.ByteString.Lazy.Char8 (ByteString)
+import Network.Freddy.Result
+
+connect :: String -> Text -> Text -> Text -> IO Connection
+connect host vhost user password =
+  openConnection host vhost user password
+
+respondTo :: Connection -> String -> (ByteString -> Either ByteString ByteString) -> IO ConsumerTag
```

Comment:
The alternative is to just expect `IO(ResultType)` from the callback which automatically forces the callback into the IO monad. This is better than what you have currently, but when I implemented this, I disliked it. 

Now to think of it, you could just provide the callback to reply to the consumer callback.
Then the consumer can do whatever it likes.